### PR TITLE
fix: changing date time to date and applying general imrpovements

### PR DESCRIPTION
**Pull request by andrey-canon**
## Description

Changes dateTime to date
Considers all the  2xx status codes
Improves response error logs

### CODE REVIEW

File: eox_nelp/signals/utils.py [hunk 1]
```diff
@@ -40,12 +40,13 @@ def _generate_external_certificate_data(timestamp, certificate_data):
     group_codes = getattr(settings, "EXTERNAL_CERTIFICATES_GROUP_CODES", {})
     course_id = str(certificate_data.course.course_key)
     extra_info = getattr(user, "extrainfo", None)
+    time = timestamp.date()
```
[the remark this answers is not in the document]
Comment (johanv26) [reply eox_nelp/signals/utils.py:43]: What is timestamp? isnt a float???

image
Or which type it is?

File: eox_nelp/signals/utils.py [hunk 2]
```diff
@@ -40,12 +40,13 @@ def _generate_external_certificate_data(timestamp, certificate_data):
...
 
     return {
         "id": certificate.id,
-        "created_at": timestamp,
+        "created_at": time,
         # Certificate doesn't have an expiration date, so this is a thing that the client must define.
-        "expiration_date": timestamp + timezone.timedelta(days=365),
+        "expiration_date": time + timezone.timedelta(days=365),
```
[the remark this answers is not in the document]
Comment (andrey-canon) [reply eox_nelp/signals/utils.py]: I  removed that in last commit based on the slack comments https://futurexlms.slack.com/archives/C04Q5CSKA8N/p1691590631770909

File: eox_nelp/signals/utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (andrey-canon) [reply eox_nelp/signals/utils.py]: timestamp renamed to time

**Review by johanv26 — APPROVED** [review]

File: eox_nelp/api_clients/certificates.py [hunk 3]
```diff
@@ -55,16 +55,16 @@ def create_external_certificate(self, certificate_data):
...
+                "name_ar": user["arabic_name"],
                 "id": user["national_id"],
                 "id_type": "saudi",
             },
             "group_code": certificate_data["group_code"],
             "certificate_type": "completion",  # What types do we have ?
             "metadata": {
                 "degree": certificate_data["grade"],
-                "FAIL": certificate_data["is_passing"],
+                "FAIL": not certificate_data["is_passing"],
```
Comment (johanv26) [inline eox_nelp/api_clients/certificates.py:67]: Ok that not has lot of logic

File: eox_nelp/signals/utils.py [hunk 4]
```diff
@@ -44,7 +44,7 @@ def _generate_external_certificate_data(time, certificate_data):
         "id": certificate.id,
         "created_at": time.date(),
         "expiration_date": None,
-        "grade": int(certificate_data.grade),
+        "grade": float(certificate_data.grade) * 100,
```
Comment (johanv26) [inline eox_nelp/signals/utils.py:47]: Does certificates receive float ? or This have to be int? Or could be float like 87.8?

**Review by johanv26 — APPROVED** [review]

File: eox_nelp/signals/utils.py (hunk 4 above)
Comment (andrey-canon) [reply eox_nelp/signals/utils.py:47]: This field was set a CharField, and  the current prod implementation shows values between  0 and 1 where  1 is  100% and  0 is 0% so there are values like 0.85

**Review by johanv26 — APPROVED** [review]

**Pull request merged**